src/database.rs

```diff
@@ -0,0 +1,41 @@
+use sqlx::{postgres::PgConnectOptions, Pool, Postgres};
+use std::env;
+use tracing::{error, info};
+
+pub async fn get_connection_pool() -> Pool<Postgres> {
+    info!("Attempting to connect to a database…");
+    let connection_pool = match connect_to_database().await {
+        Ok(connection_pool) => connection_pool,
+        Err(e) => {
+            error!("Error connecting to the database: {e}");
+            panic!();
+        }
+    };
+    info!("Connection with the database successful");
+    connection_pool
+}
+
+async fn connect_to_database() -> Result<Pool<Postgres>, sqlx::Error> {
+    let url = env::var("DATABASE_URL").expect("DATABASE_URL must be defined in .env");
+    let username =
+        env::var("DATABASE_USERNAME").expect("DATABASE_USERNAME must be defined in .env");
+    let password =
+        env::var("DATABASE_PASSWORD").expect("DATABASE_PASSWORD must be defined in .env");
+    let port_number = get_port_number();
```

Comment:
How are you getting these env vars loaded? This project doesn't use `dotenvy` or anything to get these variables...

Reply:
Besides, defining all these variables is certainly a choice when the database uri schema exists.

<img width="831" alt="image" src="https://github.com/user-attachments/assets/b185572e-ebd3-464b-a31b-899d2e8f9bcf">
<img width="988" alt="image" src="https://github.com/user-attachments/assets/b0637eba-1df9-45e7-b840-21991cf5baff">